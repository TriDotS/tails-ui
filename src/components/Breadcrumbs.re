let component = ReasonReact.statelessComponent("Breadcrumbs");

type crumb = {
  key: int,
  value: ReasonReact.reactElement,
  hasSeparator: bool
};

type crumbsArray = array(crumb);

let make = (~bgColor: string, ~crumbs: crumbsArray, ~textColor: string) => {
  ...component,
  render: _self =>
    <nav className={j|$(bgColor) p-3 rounded font-sans w-full|j}>
      <ol className="list-reset flex text-grey-dark">
        (
          crumbs
          |> Array.map((crumb: crumb) =>
               <span key=(string_of_int(crumb.key))>
                 <li className={j|$(textColor) font-bold|j}> crumb.value </li>
               </span>
             )
          |> ReasonReact.arrayToElement
        )
      </ol>
    </nav>
};

let default =
  ReasonReact.wrapReasonForJs(~component, jsProps =>
    make(
      ~bgColor=jsProps##bgColor,
      ~crumbs=jsProps##crumbs,
      ~textColor=jsProps##textColor
    )
  );
