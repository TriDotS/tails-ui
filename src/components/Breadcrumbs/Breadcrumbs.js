import PropTypes from 'prop-types'
import React from 'react'

const Breadcrumbs = ({ crumbs, bgColor, textColor }) => (
  <nav className={`${bgColor} p-3 rounded font-sans w-full m-4`}>
    <ol className="list-reset flex text-grey-dark">
      {crumbs.map(crumb => (
        <span key={crumb.key} className="flex">
          <li>
            {React.cloneElement(crumb.value, {
              className: `${textColor} font-bold`
            })}
          </li>
          {crumb.hasSeparator && (
            <li>
              <span className="mx-2">/</span>
            </li>
          )}
        </span>
      ))}
    </ol>
  </nav>
)

Breadcrumbs.defaultProps = {
  bgColor: 'bg-grey-light',
  textColor: 'text-blue'
}

Breadcrumbs.propTypes = {
  crumbs: PropTypes.array,
  bgColor: PropTypes.string,
  textColor: PropTypes.string
}

export default Breadcrumbs
