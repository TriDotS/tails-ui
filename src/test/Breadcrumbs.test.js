import Breadcrumbs from '../components/Breadcrumbs'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(
    <Breadcrumbs
      crumbs={[
        { key: 'home', value: <a href="#!">Home</a>, hasSeparator: true },
        { key: 'about', value: <a href="#!">About</a> }
      ]}
    />,
    div
  )
})

test('breadcrumbs snapshot', () => {
  const tree = renderer
    .create(
      <Breadcrumbs
        crumbs={[
          { key: 'home', value: <a href="#!">Home</a>, hasSeparator: true },
          { key: 'about', value: <a href="#!">About</a> }
        ]}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
