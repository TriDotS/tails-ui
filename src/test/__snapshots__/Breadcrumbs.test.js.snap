// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breadcrumbs snapshot 1`] = `
<nav
  className="bg-grey-light p-3 rounded font-sans w-full m-4"
>
  <ol
    className="list-reset flex text-grey-dark"
  >
    <span
      className="flex"
    >
      <li>
        <a
          className="text-blue font-bold"
          href="#!"
        >
          Home
        </a>
      </li>
      <li>
        <span
          className="mx-2"
        >
          /
        </span>
      </li>
    </span>
    <span
      className="flex"
    >
      <li>
        <a
          className="text-blue font-bold"
          href="#!"
        >
          About
        </a>
      </li>
    </span>
  </ol>
</nav>
`;
